service/controller/internal/changedetection/tccp.go

```diff
@@ -49,19 +46,22 @@ func (t *TCCP) ShouldUpdate(ctx context.Context, cr infrastructurev1alpha2.AWSCl
 	}
 
 	azsEqual := availabilityZonesEqual(cc.Spec.TenantCluster.TCCP.AvailabilityZones, cc.Status.TenantCluster.TCCP.AvailabilityZones)
-	masterInstanceEqual := cc.Status.TenantCluster.MasterInstance.Type == key.MasterInstanceType(cr)
 	operatorVersionEqual := cc.Status.TenantCluster.OperatorVersion == key.OperatorVersion(&cr)
 
 	if !azsEqual {
-		t.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("detected tenant cluster control plane should update due to availability zone changes in node pools"))
-		return true, nil
-	}
-	if !masterInstanceEqual {
-		t.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("detected tenant cluster control plane should update due to master instance type changes from %#q to %#q", cc.Status.TenantCluster.MasterInstance.Type, key.MasterInstanceType(cr)))
+		t.logger.LogCtx(ctx,
+			"level", "debug",
+			"message", "detected TCCP stack should update",
+			"reason", "availability zones changed",
```

Comment:
Introducing a `reason` field in all the change detection logs. Will be easier to digest when debugging.